### PR TITLE
feat: add `score_threshold` to Qdrant Retrievers

### LINKED ISSUE #760: Can not pass score_threshold and few other parameters to Qdrant via haystack
**Is your feature request related to a problem? Please describe.**
QdrantDocumentStore object doesn't allow me to apply parameters like score_threshold to Qdrant remote server

**Describe the solution you'd like**
Ideally the run time parameters to be passed to Qdrant could be added as kwargs to document store

**Describe alternatives you've considered**
As of now in my haystack, i have removed haystack document store and did direct integration with Qdrant. Alternative approach to filter the response after qdrant has unnecessary network performance issues.

**Additional context**
I have raised this issue in Discord and I was advised to raise feature request in Github

### PR DESCRIPTION
**Pull request by isaac-chung**
### Related Issues

- fixes #760 

### Proposed Changes:

 
 
- adding `score_threshold` to Qdrant retrievers as a class parameter and method parameter for `run()`

### How did you test it?

- Added to the unit tests for each qdrant retriever class
- Added a test case using score_threshold

### Checklist

- I have read the [contributors guidelines](https://github.com/deepset-ai/haystack-core-integrations/blob/main/CONTRIBUTING.md) and the [code of conduct](https://github.com/deepset-ai/haystack-core-integrations/blob/main/CODE_OF_CONDUCT.md)
- I have updated the related issue with new insights and changes
- I added unit tests and updated the docstrings
- I've used one of the [conventional commit types](https://www.conventionalcommits.org/en/v1.0.0/) for my PR title: `fix:`, `feat:`, `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`.

### CODE REVIEW

**Comment by isaac-chung** [comment]
Tagging @anakin87  and @julian-risch as well. I noticed that the team might be busy with other priorities, but would still appreciate some feedback. Thanks in advance!

File: integrations/qdrant/src/haystack_integrations/components/retrievers/qdrant/retriever.py [hunk 1]
```diff
@@ -48,6 +49,7 @@ def __init__(
         :param top_k: The maximum number of documents to retrieve.
         :param scale_score: Whether to scale the scores of the retrieved documents or not.
         :param return_embedding: Whether to return the embedding of the retrieved Documents.
+        :param score_threshold: A minimal score threshold for the result.
```
Comment (anakin87) [inline integrations/qdrant/src/haystack_integrations/components/retrievers/qdrant/retriever.py:52]: I would add a longer explanation. In particular, I would mention that
>  Score of the returned result might be higher or smaller than the threshold depending on the Distance function used. E.g. for cosine similarity only higher scores will be returned.

See https://python-client.qdrant.tech/qdrant_client.qdrant_client

File: integrations/qdrant/tests/test_retriever.py [hunk 2]
```diff
@@ -114,6 +118,11 @@ def test_run(self, filterable_docs: List[Document]):
         for document in results:
             assert document.embedding is None
 
+        results = retriever.run(
+            query_embedding=_random_embeddings(768), top_k=5, return_embedding=False, score_threshold=0.999
+        )["documents"]
+        assert len(results) == 0
+
```
Comment (anakin87) [inline integrations/qdrant/tests/test_retriever.py:125]: Let's introduce a specific test (`run_with_score_threshold`).
I would also suggest using a lower `score_threshold` and check it is respected by the returned docs.

**Review by anakin87 — COMMENTED** [review]
Thanks for the contribution!

I have found 2 small possible improvements.

File: integrations/qdrant/src/haystack_integrations/components/retrievers/qdrant/retriever.py (hunk 1 above)
Comment (isaac-chung) [reply integrations/qdrant/src/haystack_integrations/components/retrievers/qdrant/retriever.py:52]: Will use the suggestion.

File: integrations/qdrant/tests/test_retriever.py (hunk 2 above)
Comment (isaac-chung) [reply integrations/qdrant/tests/test_retriever.py:125]: I started with that, and realized that the random embeddings lead to changing scores. Perhaps adding a random seed might be helpful here?

Comment (anakin87) [reply integrations/qdrant/tests/test_retriever.py:125]: You can create 2 or 3 example documents with fixed embedding.
[Here is an example in Weaviate tests](https://github.com/deepset-ai/haystack-core-integrations/blob/f73c3514c13cdaaf10a878e88f5987529958e9d2/integrations/weaviate/tests/test_document_store.py#L579).

**Review by anakin87 — APPROVED** [review]
Looks good! Thank you again.

(pushed some minor fixes)

**Pull request merged**